Updated xarray methods to output dictionary

RohanSunkarapalli opened the pull request:
Added Dictionaries just like the old way method and stored the variables into it. Lastly, returned the dictionary to review the output.

RohanSunkarapalli committed:
Updated xarray methods to output dictionary

jameshalgren reviewed (changes requested):
Are there any portions of this that could be packed into a function? -- the part where we parse the variables into the dict seems like a candidate: 
``` py
for var in var_list:
    time_value = ...
```

jameshalgren commented:
ping @arpitapatel0911

jameshalgren commented:
@RohanSunkarapalli 
The timing improves in some areas, slows down in other areas. Checking this out. This is the pre-commit timing:
``` sh
Working on the old (slow) way
23.320082902908325
Working on the new way
9.199310064315796
Working on the new way with threading parallel.
9.450111865997314
Working on the new way with process parallel.
16.132300853729248
Working on the new way with loops reversed.
8.31658387184143
Working on the new way with loops reversed with threading parallel.
8.877628087997437
Working on the new way with loops reversed with process parallel.
15.884463787078857
```

Post-commit timing: 
```
Working on the old (slow) way
18.321712970733643
Working on the new way
7.316425800323486
Working on the new way with threading parallel.
8.37585186958313
Working on the new way with process parallel.
86.8927309513092
Working on the new way with loops reversed.
6.611592054367065
Working on the new way with loops reversed with threading parallel.
7.780450820922852
Working on the new way with loops reversed with process parallel.
103.29560971260071
```

RohanSunkarapalli commented:
Hello James,

I've just checked now and it seems fine to me.

First attempt -

``` sh
Working on the old (slow) way
36.05298447608948
Working on the new way
10.73807954788208
Working on the new way with threading parallel.
11.036256551742554
Working on the new way with process parallel.
25.38627552986145
Working on the new way with loops reversed.
10.309329509735107
Working on the new way with loops reversed with threading parallel.
10.118725061416626
Working on the new way with loops reversed with process parallel.
25.172446966171265
--Return--
```

Second attempt -

``` sh
Working on the old (slow) way
32.64804720878601
Working on the new way
11.237947940826416
Working on the new way with threading parallel.
11.427905082702637
Working on the new way with process parallel.
25.08159899711609
Working on the new way with loops reversed.
10.510700702667236
Working on the new way with loops reversed with threading parallel.
10.254754781723022
Working on the new way with loops reversed with process parallel.
25.02363085746765
--Return--
```


On Wed, Mar 8, 2023 at 1:41 PM James H. ***@***.***> wrote:

> @RohanSunkarapalli <https://github.com/RohanSunkarapalli>
> The timing improves in some areas, slows down in other areas. Checking
> this out. This is the pre-commit timing:
>
> Working on the old (slow) way
> 23.320082902908325
> Working on the new way
> 9.199310064315796
> Working on the new way with threading parallel.
> 9.450111865997314
> Working on the new way with process parallel.
> 16.132300853729248
> Working on the new way with loops reversed.
> 8.31658387184143
> Working on the new way with loops reversed with threading parallel.
> 8.877628087997437
> Working on the new way with loops reversed with process parallel.
> 15.884463787078857
>
> Post-commit timing:
>
> Working on the old (slow) way
> 18.321712970733643
> Working on the new way
> 7.316425800323486
> Working on the new way with threading parallel.
> 8.37585186958313
> Working on the new way with process parallel.
> 86.8927309513092
> Working on the new way with loops reversed.
> 6.611592054367065
> Working on the new way with loops reversed with threading parallel.
> 7.780450820922852
> Working on the new way with loops reversed with process parallel.
> 103.29560971260071
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/jameshalgren/data-access-examples/pull/1#issuecomment-1460751023>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AN4WNOI7B4LAYFMPJJPX24TW3DOFTANCNFSM6AAAAAAVRW3ZVE>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

jameshalgren commented:
> I've just checked now and it seems fine to me.

I was talking about running before the changes to add dictionaries and then after those changes.

RohanSunkarapalli committed:
Added comments to the updated_xarray_methods

For every new method, an empty dictionary is created called 'df_dict' which initially contains pandas dataframe for each variable in 'var_list'. The new function undergoes the method and adds the mean value of the current variable to it's corresponding  pandas dataframe. Then we close the xarray dataset and return the dictionaries containing pandas dataframe.
